app/views/ChargeSummary.scala.html

```diff
@@ -209,71 +164,71 @@ <h1 class="govuk-heading-xl">
         }
     }
 
-    @if(viewModel.codingOutEnabled && documentDetail.isPayeSelfAssessment){
+    @if(viewModel.codingOutEnabled && viewModel.documentDetail.isPayeSelfAssessment){
         @checkPayeCodePara
         <div id="coding-out-wrapper">
             @govukInsetText(InsetText(content = HtmlContent(codingOutMessageInset)))
         </div>
         @codingOutMessage
     }
 
-    @if(viewModel.codingOutEnabled && documentDetail.isCancelledPayeSelfAssessment){
+    @if(viewModel.codingOutEnabled && viewModel.documentDetail.isCancelledPayeSelfAssessment){
         @checkPayeCodePara
         <div id="cancelled-coding-out-wrapper">
             @govukInsetText(InsetText(content = HtmlContent(codingOutCancelledMessageInset)))
         </div>
         @codingOutMessage
     }
 
-    @if(viewModel.codingOutEnabled && documentDetail.isClass2Nic && !documentDetail.isPayeSelfAssessment){
+    @if(viewModel.codingOutEnabled && viewModel.documentDetail.isClass2Nic && !viewModel.documentDetail.isPayeSelfAssessment){
```

Comment:
here is commit/change example how this can be refactored into view model property/logic: https://github.com/hmrc/income-tax-view-change-frontend/pull/2412/commits/8fdcbdf620e212a50a091f3f8599119e9e9c7538